lib.php

```diff
@@ -77,15 +77,26 @@ function observation_add_instance($data): int {
  * @return bool true on success, false or a string error message on failure.
  */
 function observation_update_instance($data): bool {
-    return \mod_observation\observation_manager::modify_instance(array(
+    $success = \mod_observation\observation_manager::modify_instance(array(
         "id" => $data->instance,
         "name" => $data->name,
         "timemodified" => time(),
         "observer_ins" => $data->observerins_editor['text'],
         "observer_ins_f" => $data->observerins_editor['format'],
         "observee_ins" => $data->observeeins_editor['text'],
         "observee_ins_f" => $data->observeeins_editor['format'],
+        "students_self_unregister" => (int) $data->students_self_unregister
     ));
+
+    // Update all the calendar events to get the new data.
```

Comment:
Should this be locked behind `if ($success)` incase of a weird failure condition?